core/src/apps/common/request_pin.py

```
@@ -113,27 +138,63 @@
+                verified = result
+                usertype = PinResult.USER_PIN_ENTERED
+            print("usertype", usertype)
+
+            if verified:
+                if usertype == PinResult.PASSPHRASE_PIN_ENTERED:
+                    device.set_passphrase_pin_enabled(True)
+                elif usertype == PinResult.USER_PIN_ENTERED:
+                    device.set_passphrase_pin_enabled(False)
+
+        except Exception:
+            raise wire.PinCancelled("cancle")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)